src/vtt-segment-loader.js

```diff
@@ -361,7 +361,9 @@ export default class VTTSegmentLoader extends SegmentLoader {
     segmentInfo.cues.forEach((cue) => {
       // remove any overlapping cues to prevent doubling
       this.remove(cue.startTime, cue.endTime);
-      this.subtitlesTrack_.addCue(cue);
+      this.subtitlesTrack_.addCue(this.hls_.player_.tech_.featuresNativeTextTracks ?
```

Comment:
So, it looks like the mock config that is used in the tests does not have the `hls_` property.

We could either
1. Add this to the mock
2. Check the full path to the `featuresNativeTextTracks` property (see below)

```js
this.hls_
  && this.hls_.player_
  && this.hls_.player_.tech_
  && this.hls_.player_.tech_.featuresNativeTextTracks
```
@gkatsev do you have a preference for which? Should we also add some tests for when this is `true` vs `false`?

Reply:
After some discussion, a better way will likely to pass a `featuresNativeTextTracks` flag when we create the `VTTSegmentLoader` in the `master-playlist-controller` https://github.com/videojs/http-streaming/blob/450ca91ade3c8fd7b96834e811ccd6699daee834/src/master-playlist-controller.js#L172-L175
Adding something like:
```js
featuresNativeTextTracks: this.tech_.featuresNativeTextTracks
```
as an option next to `loaderType`. Then setting is a local option in the VTTSegmentLoader constructor https://github.com/videojs/http-streaming/blob/86dfba96a3cb3f812caae2bcf25bd40365572221/src/vtt-segment-loader.js#L36